x-pack/plugins/security_solution/server/lib/entity_analytics/risk_engine/get_user_risk_engine_privileges.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { KibanaRequest } from '@kbn/core/server';
+import type { SecurityPluginStart } from '@kbn/security-plugin/server';
+import type { RiskEnginePrivilegesResponse } from './types';
+import {
+  RISK_ENGINE_REQUIRED_ES_CLUSTER_PRIVILEGES,
+  RISK_ENGINE_REQUIRED_ES_INDEX_PRIVILEGES,
+} from '../../../../common/risk_engine';
+
+const groupByPrivilege = (
+  privileges: Array<{
+    privilege: string;
+    authorized: boolean;
+  }>
+): Record<string, boolean> => {
+  return privileges.reduce<Record<string, boolean>>((acc, { privilege, authorized }) => {
+    return {
+      ...acc,
```

Comment:
Since they can potentially be hot code paths, we tend to avoid these unnecessary object creations within a reduce loop like this. [Here](https://github.com/elastic/kibana/pull/170922) is a recent example.

In this situation, of course, the expected number of privileges should be incredibly small, but it seems like we're just avoiding that pattern altogether now due to its potential for problems.

Reply:
Love it 👌